### PR TITLE
feat: (2) Solargraph Remote Debugging with Docker

### PR DESCRIPTION
**Pull request by tanmoysrt**
Fixes #3761

#### Describe the changes you have made in this PR -
- Modified Dockerfile to install `solargraph`
- Modified docker-compose.yml to expose port 3002
- Modified docker_run script to run solargraph inside Docker

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -135,6 +135,9 @@ group :development, :test do
   gem "rubocop-rspec", require: false
   gem "rbs_rails"
   gem "steep"
+  # Adds support for Solargraph LSP
+  gem 'solargraph-rails', '~> 0.3.1'
+  gem "undercover"
```
Comment (tachyons) [inline Gemfile:140]: Unrelated ?

Comment (tanmoysrt) [reply Gemfile:140]: yes, while resolving conflicts, this has come by mistake
will remove that
